### DISCUSSION TITLE
Uhh... Help?

### DISCUSSION THREAD

**Discussion by MoisesHernandezSanchez** [mathoudebine/turing-smart-screen-python#347] [Q&A] [answered]
Sooo... after my issue with Pillow (Solved by updating Python to 3.11.6, NOT 3.12), I ran the configuration, works like a charm. Ran Main, my screen shows nothing but the tray Icon shows. Here's the log of one run both from start to run to after I exit out:

10/3/2023 11:47:35 AM [INFO] Loading theme 3.5inchTheme2 from res/themes/3.5inchTheme2/theme.yaml
10/3/2023 11:47:35 AM [DEBUG] HW revision: B
10/3/2023 11:47:35 AM [DEBUG] Static COM port: COM1
10/3/2023 11:47:36 AM [DEBUG] Found LibreHardwareMonitorLib 0.9.2.0
10/3/2023 11:47:37 AM [INFO] Found CPU: AMD Ryzen 5 3600
10/3/2023 11:47:37 AM [INFO] Found Memory: Generic Memory
10/3/2023 11:47:37 AM [INFO] Found Nvidia GPU: NVIDIA GeForce RTX 3060 Ti
10/3/2023 11:47:37 AM [INFO] Found Storage: WDC WD10EZRZ-00HTKB0
10/3/2023 11:47:37 AM [INFO] Found Storage: WDC WDBNCE2500PNC
10/3/2023 11:47:37 AM [INFO] Found Storage: KINGSTON SA400S37120G
10/3/2023 11:47:37 AM [INFO] Found Network interface: Bluetooth Network Connection
10/3/2023 11:47:37 AM [INFO] Found Network interface: Main
10/3/2023 11:47:37 AM [INFO] Found Network interface: vEthernet (Main)
10/3/2023 11:47:37 AM [DEBUG] Using Python 3.11.6 (tags/v3.11.6:8b6ee5b, Oct  2 2023, 14:57:12) [MSC v.1935 64 bit (AMD64)]
10/3/2023 11:47:37 AM [INFO] Tray icon has been displayed
10/3/2023 11:48:31 AM [DEBUG] Caught Windows control event 0, exiting
10/3/2023 11:48:31 AM [INFO] Your display does not support custom brightness level
10/3/2023 11:48:31 AM [INFO] Only HW revision 'flagship' supports backplate LED color setting
10/3/2023 11:48:31 AM [INFO] Waiting for all pending request to be sent to display (5s max)...
10/3/2023 11:48:37 AM [DEBUG] (5.1s)

I thought maybe changing the brightness down to 0% would work but it didn't help. Same exact log. It shows in the system tray just fine, but nothing shows on the screen.
Any ideas? And sorry for all the questions.

**Comment by mathoudebine** [comment]
From the logs I can see your display does not answer to the commands that are sent to detect its model.
Are you sure your screen really is a XuanFang? Please check this page to confirm: https://github.com/mathoudebine/turing-smart-screen-python/wiki/Hardware-revisions
If the model is correct, you may have selected the wrong COM port during the Configuration. Run it again and select another COM port (or AUTO but it may not always work with rev. B)

**Reply by MoisesHernandezSanchez** [reply]
Hi, sorry for the late reply. Based on the image, I can conclude that my XuanFang is actually a Rev. "flagship" due to the presence of a backplate, but the log seems to state that the program believes it to be a Rev. B instead. The configuration doesn't allow me to select which. What should I do?

**Reply by MoisesHernandezSanchez** [reply]
OK SO IMPORTANT: It works now. I don't know what I did or why it's working now. I did try COM3 which failed last time but this time works. I'm sorry I can't give any info on why it works now since I'm 90% sure I did nothing new, but I suppose COM3 felt cooperative now.
Here's the log from it:
10/3/2023 5:28:25 PM [INFO] Loading theme LandscapeEarth from res/themes/LandscapeEarth/theme.yaml
10/3/2023 5:28:25 PM [DEBUG] HW revision: B
10/3/2023 5:28:25 PM [DEBUG] Static COM port: COM3
10/3/2023 5:28:26 PM [DEBUG] Found LibreHardwareMonitorLib 0.9.2.0
10/3/2023 5:28:27 PM [INFO] Found CPU: AMD Ryzen 5 3600
10/3/2023 5:28:27 PM [INFO] Found Memory: Generic Memory
10/3/2023 5:28:27 PM [INFO] Found Nvidia GPU: NVIDIA GeForce RTX 3060 Ti
10/3/2023 5:28:27 PM [INFO] Found Storage: WDC WD10EZRZ-00HTKB0
10/3/2023 5:28:27 PM [INFO] Found Storage: WDC WDBNCE2500PNC
10/3/2023 5:28:27 PM [INFO] Found Storage: KINGSTON SA400S37120G
10/3/2023 5:28:27 PM [INFO] Found Network interface: Bluetooth Network Connection 2
10/3/2023 5:28:27 PM [INFO] Found Network interface: Main
10/3/2023 5:28:27 PM [INFO] Found Network interface: vEthernet (Main)
10/3/2023 5:28:27 PM [DEBUG] Using Python 3.11.6 (tags/v3.11.6:8b6ee5b, Oct  2 2023, 14:57:12) [MSC v.1935 64 bit (AMD64)]
10/3/2023 5:28:27 PM [INFO] Tray icon has been displayed
10/3/2023 5:28:28 PM [DEBUG] HW sub-revision: 2577
10/3/2023 5:28:28 PM [INFO] Only HW revision 'flagship' supports backplate LED color setting
10/3/2023 5:28:28 PM [DEBUG] Drawing Image: BACKGROUND
10/3/2023 5:28:28 PM [DEBUG] Bitmap res/themes/LandscapeEarth/background.png is now loaded in the cache
10/3/2023 5:28:28 PM [DEBUG] Found one supported GPU: NVIDIA GeForce RTX 3060 Ti

I also realize I'm going to want to move it out of the downloads folder now. Thanks for the help and sorry I couldn't help unravel the mystery other than "solved somehow"

**Reply by mathoudebine** [reply, marked as answer]
Rev. B and rev. flagship are managed the same way from the program. During the configuration just select:
![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/38615348/a86aaac8-b952-4736-bbeb-49e57c740b85)
The program will automatically detect if your screen is rev. b or rev. flasghip when running.

I see you now have resolved your issue, nice work! Sometimes the screen may be stuck, in this case you may have to unplug it then plug it back in to fully reset it (or reboot the computer)

**Reply by MoisesHernandezSanchez** [reply]
Yeah, sorry for the bother. Still, thanks for going out of the way to keep responding!
